src/prof_data.c

```diff
@@ -1037,6 +1037,12 @@ prof_leakcheck(const prof_cnt_t *cnt_all, size_t leak_ngctx) {
 		    1) ? "s" : "", leak_ngctx, (leak_ngctx != 1) ? "s" : "");
 		malloc_printf(
 		    "<jemalloc>: Run jeprof on dump output for leak detail\n");
+		if (opt_prof_leak_error) {
+			malloc_printf(
+			    "<jemalloc>: Exiting with error code because of whole-"
+			    "program memory leaks\n");
```

Comment:
What does `whole-program` mean exactly? What are the other types of memory leaks?

Reply:
whole-program means **bin** compile witch jemalloc.
However, in tcmalloc, we can check whole-program heap witch several levels，such as :
1. minimal
2. normal
3. strict
4. draconian
From: https://gperftools.github.io/gperftools/heap_checker.html
Does this exist in jemalloc？Thanks a lot for your answer.

Reply:
No, we don't have that. I think what we currently have is classified as "draconian" in that document :) We also don't rely on "live objects" to detect leaked memory, we simply report everything that was malloced and sampled, but wasn't freed before atexit(). If you use C++, this might be important for you, as global objects are destroyed after atexit, so you can see false positives this way.

Reply:
> 

Got it, thanks for your reply